### PR TITLE
synthetics(mobile): Catch invalid application size and add more context in the error message

### PR DESCRIPTION
**Pull request by KevinYeramian**
### What and why?

The synthetics mobile products allow customers to use `mobileApplicationVersionFilePath` option to run tests with custom local application. This option implies an upload to our server via a pre-signed URL. We limit the size of the application to avoid misusage from 1KiB to 1GiB. 

Before this PR, datadog-ci command stops with:

image

This PR adds more context to the error:

image

### How?

Add a verification of the size before uploading the application via the pre-signed URL. 

### Review checklist

- [x] Feature or bugfix MUST have appropriate tests (unit, integration)

### CODE REVIEW

**Review by ClaraJacintho — APPROVED** [review]
Looks good for Mobile 👍🏻

File: src/commands/synthetics/__tests__/mobile.test.ts [hunk 1]
```diff
@@ -165,7 +194,11 @@ describe('shouldUploadApplication', () => {
   })
 })
 
-describe('overrideMobileConfig', () => {
+describe('overrideMobileConfig 1', () => {
```
Comment (ypetit) [inline src/commands/synthetics/__tests__/mobile.test.ts:197]: What is this `1` you added to to the name test used for ?

File: src/commands/synthetics/mobile.ts [hunk 2]
```diff
@@ -26,6 +30,14 @@ export const uploadMobileApplications = async (
   )
 
   const fileBuffer = await fs.promises.readFile(applicationPathToUpload)
+  const fileBufferBytesSize = fileBuffer.byteLength
+  if (fileBufferBytesSize < MIN_APPLICATION_SIZE || MAX_APPLICATION_SIZE < fileBufferBytesSize) {
+    throw new CriticalError(
+      'INVALID_MOBILE_APPLICATION_SIZE',
+      `Invalid Mobile Application size. Expect a size between ${MIN_APPLICATION_SIZE}B and ${MAX_APPLICATION_SIZE}B, got ${fileBufferBytesSize}B.`
```
Comment (ypetit) [inline src/commands/synthetics/mobile.ts:37]: Is it really complicated to format in a human readable way the size displayed ? To avoid the `1073741824B`

**Review by ypetit — APPROVED** [review]

**Review by jakepruitt — APPROVED** [review]
Looks good to me

**Pull request closed without merging**